Python/Product/VSInterpreters/Interpreter/ICondaLocatorProvider.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+namespace Microsoft.PythonTools.Interpreter {
+
+    public interface ICondaLocatorProvider {
```

Comment:
Difference between this and `ICondaLocator`?

Reply:
There's one instance of the provider, but multiple instances of locator.

Provider figures out which locator to use. This is based on priority, from highest to lowest:
Tools/Option/Python/Conda path > Bundled miniconda path > latest installed Anaconda/Miniconda path

You are making me realize that I should have the locator provider return a locator instead of returning the conda path of the appropriate locator. That would make it clearer. I'll make the change and add comments too.